packages/nodes-base/credentials/SpotifyOAuth2Api.credentials.ts

```diff
@@ -0,0 +1,53 @@
+import {
+	ICredentialType,
+	NodePropertyTypes,
+} from 'n8n-workflow';
+
+
+export class SpotifyOAuth2Api implements ICredentialType {
+	name = 'spotifyOAuth2Api';
+	extends = [
+		'oAuth2Api',
+	];
+	displayName = 'Spotify OAuth2 API';
+	properties = [
+		{
+			displayName: 'Spotify Server',
+			name: 'server',
+			type: 'hidden' as NodePropertyTypes,
+			default: 'https://api.spotify.com/',
+		},
+		{
+			displayName: 'Authorization URL',
+			name: 'authUrl',
+			type: 'hidden' as NodePropertyTypes,
+			default: 'https://accounts.spotify.com/authorize',
```

Comment:
Should abstract into BASE_URL template literal

Reply:
What do you mean by this?

Reply:
Something like `${BASE_URL}/authorize`.

Reply:
Or something more flexible for sub domains could also be useful like injecting {accounts | songs}.spotify.com
This is just extra though not a requirement

Reply:
Check it out here:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

It's a new thing in es6